src/Game/GamePage.js

```diff
@@ -42,17 +37,19 @@ const GamePage = (props) => {
     let mySubscription = supabase
       .from("Players")
       .on("UPDATE", (payload) => {
-        const { totalPlayersHp } = stateRef.current;
-
-        const changeHitpoints = totalPlayersHp.map((playerData) => {
-          if (payload.new.name === playerData.name) {
-            return { name: payload.new.name, Hitpoints: payload.new.Hitpoints };
-          } else {
-            return { name: playerData.name, Hitpoints: playerData.Hitpoints };
-          }
+        setTotalPlayersHp((currentState) => {
+          const changeHitpoints = currentState.map((playerData) => {
+            if (payload.new.name === playerData.name) {
+              return {
+                name: payload.new.name,
+                Hitpoints: payload.new.Hitpoints,
+              };
+            } else {
+              return { name: playerData.name, Hitpoints: playerData.Hitpoints };
+            }
+          });
+          return changeHitpoints;
```

Comment:
this code is getting very nested and hard to read so lets move the callback function to its own named function outside